### PR TITLE
feat(rust, python): add utc parameter to strptime

### LINKED ISSUE #6466: utc parameter in strptime
### Problem description

When processing multiple fixed offsets / timezones, pandas has a `utc` parameter which allows you to still parse the elements, and then set whatever timezone you want.

For example:
```python
In [73]: ts = ['2021-03-27T23:59:59+01:00', '2021-03-28T23:59:59+02:00']

In [74]: pd.to_datetime(ts, utc=True).tz_convert('Europe/Vienna')
Out[74]: DatetimeIndex(['2021-03-27 23:59:59+01:00', '2021-03-28 23:59:59+02:00'], dtype='datetime64[ns, Europe/Vienna]', freq=None)
```

I don't think there's a way to do this in polars, other than
```python
In [76]: pl.Series(ts).str.split('+').arr.get(0).str.strptime(pl.Datetime).dt.with_time_zone('UTC').dt.cast_time_zone('Europe/Vienna')
Out[76]:
shape: (2,)
Series: '' [datetime[μs, Europe/Vienna]]
[
        2021-03-27 23:59:59 CET
        2021-03-28 23:59:59 CEST
]
```

I'll include this as a recipe in https://github.com/pola-rs/polars-book/pull/234, but longer-term, would you be open to a UTC parameter?

- `utc=True` would convert each element to UTC and set that as the timezone (like pandas does)
- `utc=False` with mixed timezones would error (unlike pandas - but [my hope is that this is what pandas will do too](https://github.com/pandas-dev/pandas/issues/50887))

---

[SO question for reference](https://stackoverflow.com/questions/75243040/conversion-of-iso-datetime-string-to-pl-datetime-failure/75246248#75246248)

### PR DESCRIPTION
**Pull request by MarcoGorelli**
closes #6466 

demo:
```python
In [3]: ts = ['2021-03-27T23:59:59+01:00', '2021-03-28T23:59:59+02:00']

In [4]: pd.Series(pd.to_datetime(ts, utc=True))
Out[4]: 
0   2021-03-27 22:59:59+00:00
1   2021-03-28 21:59:59+00:00
dtype: datetime64[ns, UTC]

In [5]: pl.Series(ts).str.strptime(pl.Datetime, '%+', utc=True)
Out[5]: 
shape: (2,)
Series: '' [datetime[μs, UTC]]
[
        2021-03-27 22:59:59 UTC
        2021-03-28 21:59:59 UTC
]
```

### CODE REVIEW

File: polars/polars-time/src/chunkedarray/utf8/mod.rs [hunk 1]
```diff
@@ -421,13 +427,15 @@ pub trait Utf8Methods: AsUtf8 {
...
-                                    return Err(PolarsError::ComputeError(
-                                        "Different timezones found during 'strptime' operation.".into(),
-                                    ));
+                        if !utc {
+                            match tz {
+                                None => tz = Some(dt.timezone()),
+                                Some(tz_found) => {
+                                    if tz_found != dt.timezone() {
+                                        return Err(PolarsError::ComputeError(
+                                            "Different timezones found during 'strptime' operation.".into(),
```
Comment (FObersteiner) [inline polars/polars-time/src/chunkedarray/utf8/mod.rs:436]: This error might now hint to using `utc=True`

**Review by FObersteiner — APPROVED** [review]
Looks good to me, thank you for implementing this :) 

A small detail you might add: see my comment on the ComputeError.

File: polars/polars-time/src/chunkedarray/utf8/mod.rs (hunk 1 above)
Comment (MarcoGorelli) [reply polars/polars-time/src/chunkedarray/utf8/mod.rs:436]: good one, thanks!

**Comment by ritchie46** [comment]
Thanks a lot @MarcoGorelli  and @FObersteiner. Great addition. :)

**Pull request merged**